packages/render-fidelity-tools/src/artifact-creator.ts

```diff
@@ -48,168 +48,186 @@ export class ArtifactCreator {
+    }
+    const screenshotImage = pngjs.PNG.sync.read(screenshot).data;
 
-    const finalConfig: ImageComparisonConfig =
-        Object.assign({}, this.config, {scenarios: analyzedScenarios});
+    const modelViewerIndex = 0;
+    const modelViewerGoldenPath = join(
+        rootDirectory, 'goldens', scenarioName, goldens[modelViewerIndex].file);
+    let modelViewerGolden;
+    try {
+      modelViewerGolden = await fs.readFile(modelViewerGoldenPath);
+    } catch (error) {
+      throw new Error(`❌ Failed to read model-viewer's ${
+          scenarioName} golden! Error message: ${error.message}`);
+    }
+    const modelViewerGoldenImage = pngjs.PNG.sync.read(modelViewerGolden).data;
 
-    await fs.writeFile(
-        join(outputDirectory, 'config.json'), JSON.stringify(finalConfig));
+    const result =
+        await this.analyze(screenshotImage, modelViewerGoldenImage, dimensions);
 
-    await fs.writeFile(
-        join(outputDirectory, 'modelViewerFidelityErrors.json'),
-        JSON.stringify(modelViewerFidelityErrors));
+    const rmsInDb = toDecibel(result.rmsDistanceRatio);
 
-    await fs.writeFile(
-        join(outputDirectory, 'modelViewerFidelityWarnings.json'),
-        JSON.stringify(modelViewerFidelityWarnings));
+    if (rmsInDb > FIDELITY_TEST_THRESHOLD) {
+      throw new Error(`❌ Senarios name: ${scenario.name}, rms distance ratio: ${
+          rmsInDb.toFixed(2)} dB.`);
+    }
 
-    return scenarios;
+    return result;
   }
 
-  protected async analyze(
-      screenshot: Buffer, goldens: Array<GoldenConfig>,
-      scenario: ScenarioConfig,
-      dimensions: Dimensions): Promise<AnalysisResults> {
-    const analysisResults: AnalysisResults = [];
-    const {rootDirectory, outputDirectory} = this;
-    const {name: scenarioName, exclude} = scenario;
+  async fidelityTest(scenarioWhitelist: Set<string>|null = null) {
+    const {scenarios} = this.config;
+    const {outputDirectory} = this;
+    const analyzedScenarios: Array<ScenarioConfig> = [];
+    const autoTestResults:
+        AutoTestResults = {results: [], errors: [], warnings: []};
 
-    for (const goldenConfig of goldens) {
-      const {name: rendererName} = goldenConfig;
+    for (let scenarioBase of scenarios) {
+      const scenarioName = scenarioBase.name;
+      const scenario = this[$configReader].scenarioConfig(scenarioName)!;
 
-      if (exclude != null && exclude.includes(rendererName)) {
+      if (scenarioWhitelist != null && !scenarioWhitelist.has(scenarioName)) {
         continue;
       }
 
-      console.log(
-          `\n🔍 Comparing <model-viewer> to ${goldenConfig.description}`);
+      console.log(`\n🎨 Scenario: ${scenarioName}`);
 
-      const goldenPath =
-          join(rootDirectory, 'goldens', scenarioName, goldenConfig.file)
+      const scenarioOutputDirectory = join(outputDirectory, scenarioName);
+      mkdirp.sync(scenarioOutputDirectory);
+
+      await this.compareRenderers(scenario);
```

Comment:
im not using try catch block for comapre renderers because i feel like auto test has already most possible errors. But compareRenderers probably may have its own errors as well, such as it can't find some renderers' goldens. If you want me to try and catch these errors as well im happy to change.